### ISSUE TITLE
function return data types

### ISSUE THREAD

**Issue by walkerml** [MDSplus/mdsplus#2]
Why do general purpose mdsplus data functions for matlab return non-general data types?  For example,  mdsconnect returns status having type MDSplus.Int32 rather than ordinary "int", mdsvalue returns java.lang.String rather than ordinary "string" when data is not found.  Many standard matlab functions do not recognize these data types, which then requires writing special wrappers to convert these non-standard types into standard types.  Why don't these functions just return standard types?

**Comment by joshStillerman** [comment]
Hi -
  The matlab functions mds*() are supposed to return standard types.  Connect may need fixing.  The string returns have been fixed as far as I know.  There is one further change to the string code when 2d arrays of strings are referenced.  Do you have the current (alpha) /usr/local/mdsplus/matlab  ?  If so, and you have specific things that are not working correctly, let me know.

-Josh

**Comment by hahnsh** [comment]
Hi,
I have had the same issue reported in this thread, when I used a version 6.1.26 at a Linux server (SL6.3). Switching to the current  "alpha" version on Mar/18/2015 resolved my problem, but my test might be too simple to answer the original question:

I tested the standard mds function at matlab R2014b as
```
mds_server = getenv('KSTAR_PCS_MDS_SERVER');
mdsconnect(mds_server);
status = mdsopen('kstar',shot);
  [str,ier] = mdsvalue('\T0_STR')
```
\TO_STR is supposed to read as a string. 
If I use old version (MDS+ 6.1.26) for this command, mdsvalue() returns 
```
  str= [1x1 java.lang.String]
```
which is not good. If I switched to the alpha, 
```
  str =  '2014/08/11 14:06:37.898561'
```
FYI: Browsing the history on malab/NATIVEvalue.m, I found the fix has been applied quite recently so I suspect any "stable" version available at mdsplus.org does not reflect the fix.
